src/runtime_src/core/include/xcl_api_macros.h

```diff
@@ -1022,3 +1022,27 @@ mtx.unlock();
     xclLoadXclbinContent_SET_PROTO_RESPONSE(); \
     FREE_BUFFERS(); \
     xclLoadXclbinContent_RETURN();
+
+#define xclPfBarAddrmap_SET_PROTOMESSAGE(func_name,req_val) \
+    c_msg.set_req(req_val); \
+
+#define xclPfBarAddrmap_RPC_CALL(func_name,req_val,pfbarmap_vec) \
+  RPC_PROLOGUE(func_name); \
+  xclPfBarAddrmap_SET_PROTOMESSAGE(func_name,req_val); \
+  SERIALIZE_AND_SEND_MSG(func_name) \
+  xclPfBarAddrmap_SET_PROTO_RESPONSE(pfbarmap_vec); \
+  FREE_BUFFERS();
+
+#define xclPfBarAddrmap_SET_PROTO_RESPONSE(pfbarmap_vec) \
+  if(r_msg.pfbar_map_size() != 0)\
+  {\
+    for(int i = 0; i < r_msg.pfbar_map_size() ; i++) \
+    { \
+      PF_BAR_st_type *ptr = new PF_BAR_st_type; \
```

Comment:
I dont see destructor of this object.

Reply:
Now all dynamically allocated PF_BAR_st_type  objected are getting deleted in destructor.